### DISCUSSION TITLE
version 24.3

### DISCUSSION THREAD

**Discussion by hvianna** [hvianna/audioMotion.js#51] [Announcements]
<a href="https://audiomotion.app" target="_blank" title="open web app"><img src="https://img.shields.io/badge/audioMotion.app-ff3e37?style=flat" height="48"></a>

📢 **audioMotion** is now an [online web app](https://audiomotion.app) that can play **music and video** directly from your device! No download necessary.

### Added:

+ Support for the File System Access API, allowing the web app to read files from your local device (depends on [browser support](known-issues.md));
+ **Video playback** - [suggested](https://github.com/hvianna/audioMotion.js/discussions/40) by **@JesusGod-Pope666-Info**;
+ *Bark*, *Mel* and linear [**frequency scales**](users-manual.md#frequency-scale) (\*\*) - suggested by **@TF3RDL** ([here](https://github.com/hvianna/audioMotion-analyzer/issues/30)) and **@hsnam95** ([here](https://github.com/hvianna/audioMotion.js/issues/34));
+ [Frequency **weighting filters**](users-manual.md#weighting) (\*\*);
+ New [dual-channel layouts](users-manual.md#channel-layout): horizontal (side-by-side channels) and combined (overlaid channels) - [suggested](https://github.com/hvianna/audioMotion-analyzer/issues/38) by **@TF3RDL**;
+ Allow selecting **different [gradients](users-manual.md#gradients) for each channel,** when using a dual-channel layout;
+ New [**bar coloring modes**](users-manual.md#color-mode): by level (bar amplitude) and by index (bar position);
+ Press **0** (zero) to quickly randomize analyzer settings;
+ [**ANSI** switch](users-manual.md#switches) to use IEC/ANSI preferred frequencies for octave bands - [suggested](https://github.com/hvianna/audioMotion.js/issues/28) by **@jonathan-annett**;
+ [**LINEAR** switch](users-manual.md#switches) to use **linear amplitude** values, instead of decibels (\*\*);
+ [**NOTES** switch](users-manual.md#switches) to display **musical note labels** in the X-axis scale;
+ [**ROUND** switch](users-manual.md#switches) to render analyzer bars wih **rounded corners** on top;
+ [Config options](users-manual.md#general-settings) to customize the **backgrounds folder** and how many individual media files appear in the Background options;
+ [Config option](users-manual.md#general-settings) to limit the maximum frame-rate;
+ [Config option](users-manual.md#general-settings) to adjust on-screen display font size - [suggested](https://github.com/hvianna/audioMotion.js/issues/44) by **@kimycai**;
+ [Config option](users-manual.md#general-settings) to **auto-hide the media panel**, expanding the analyzer area;
+ [Config options](users-manual.md#general-settings) to remember the contents of the play queue and the last used folder between sessions;
+ Enabled some server-side configuration options via `config.json` file - see [Self-hosting audioMotion](server.md);

\*\* Special thanks to **@TF3RDL** for the [code snippets](https://github.com/hvianna/audioMotion-analyzer/issues/30) which helped immensely in the implementation of these features!

### Changed & improved:

+ User preset funcionality now provides 9 slots for saving your favorite analyzer configurations;
+ **Prism** and **Rainbow** gradients have been updated with softer colors (the old versions can be enabled in the Config panel);
+ Random Mode has been renamed to **Randomize** and it's now possible to remove the *Analyzer Mode* from the randomized settings;
+ **AUTO** gradient functionality has been changed to *Gradients* option under [Settings affected by Randomize](users-manual.md#settings-affected-by-randomize);
+ **PEAKS** now work for *Line* and *Area graph* analyzer modes in *Radial* view;
+ **STEREO** switch has been replaced by the **Channel Layout** setting;
+ Peaks hold and decay times are now more consistent on varying frame rates;
+ Improved frequency scale labeling (label font size slightly reduced in fullscreen);
+ Play queue size limit increased to 2000 files;
+ **Smoothing** setting has been moved to [General settings](users-manual.md#general-settings);
+ Added more options to the minimum and maximum values in the [Frequency range](users-manual.md#frequency-range);
+ New default settings: 20Hz - 20kHz frequency range, FFT smoothing set to 0.7, Scale-Y off and info displayed at both the start and end of the song;
+ New fonts for the user interface and internal console;
+ New custom UI controls to replace standard range inputs and radio buttons;
+ If you are [self-hosting audioMotion](server.md), it can now run from any subdirectory (no need to use a virtual host or custom port);
+ [Updated documentation](https://audiomotion.app/docs);

### Fixed:

+ Reflex not working on PIP mode on some circumstances;
+ Some special characters in file/directory names preventing files from being loaded.

---

Release: https://github.com/hvianna/audioMotion.js/releases/tag/v24.3

**Comment by JesusGod-Pope666-Info** [comment]
I certainly still do not like the font letters used.

And I have some issues with the scrollbar and the app doing sort of weird things - I guess it is this autohide thing which I am not sure I like.... Like a scroll bar is really not that big you know....

A side thought, if you ever have time, a mobile version would be nice as well of the player but would be a project in itself I guess.

**Reply by hvianna** [reply]
Yea, mobile is a big challenge with so many stuff on screen.
You can disable the auto-hide in the Config panel, but the scrollbar issue is unintentional. Maybe it's a browser-specific issue. I'll have to look into that.

**Comment by JesusGod-Pope666-Info** [comment]
I guess you can only add 2000 songs.... I guess it is enough, but.... Okay. Just doing some testing and all and had a bunch of files and wanted to see what happened. I guess you would usually not add 2000 songs.

**Comment by JesusGod-Pope666-Info** [comment]
Maybe some grey or sorta little different color for the file list, shifting from the two colors, for easier seeing discerning the files in the list, don't need to be much difference I think might..... Maybe not... I just thought about it.

**Comment by JesusGod-Pope666-Info** [comment]
Ahhh I guess I will be able to config some of the things when the new feature comes about, the scrollbar, will most likely be one thing I turn off :P

**Reply by hvianna** [reply]
There shouldn't be a scrollbar.. can you post a screenshot of that?
Which browser are you using and what's your screen resolution?

**Reply by JesusGod-Pope666-Info** [reply]
Kinda need a scrollbar for my low resolution computer. :) I am not on something hugely modern, well a Crapbar. 1280*720.
Like I do my videos usually at a maximum of 1280*720 anyhow, less space and less processing or graphic power for other users that does not have the newest of the newest.

**Reply by JesusGod-Pope666-Info** [reply]
Kinda hard to just take an image and see.... As it is kinda of a flow video thing in action.
I think it is with all browsers, and probably able to reproduce it with a lower resolution where the player don't exactly fit.
Although even if it does not fit exactly do my screen I usually get around it with the scrollbar that now disappears in the new version and then comes up doing some weird stuff.

**Reply by hvianna** [reply]
@JesusGod-Pope666-Info Understood. I'll fix it asap.

**Comment by JesusGod-Pope666-Info** [comment]
might be nice to be able to choose a folder even if it is loading.... As if I rename a folder and it loads, it just stalls..... I guess a reset of the cache of the browser might work. But I can't push the loading button and point out another folder which actually exist.

**Reply by hvianna** [reply]
I'll try to improve that! ✍️
